sonic_installer/main.py

```diff
@@ -938,7 +939,8 @@ def rollback_docker(container_name):
     version_tag = ""
     for id in image_id_all:
         if id != image_id_previous:
-            version_tag = get_docker_tag_name(id)
```

Comment:
>get_docker_tag_name [](http://example.com/codeflow?start=26&length=19)

you can change the implementation of `get_docker_tag_name()`, and this function should be easy to unit test, you still need some mock.

Reply:
Hi qi, thanks for the suggestion. I would to rollback this change because I found the real issue is in sonic build system. Here is my analyze:

During my previous manual test, I found `get_docker_tag_name()` cannot return the correct tag name of docker.  The return value of `get_docker_tag_name()` does not equal to output of `docker images`.

However, I re-visit the logic today, and I found:
1. `get_docker_tag_name()` tries to get the docker tag by querying a docker label specified by sonic make file.
```
cmd = "docker inspect --format '{{.ContainerConfig.Labels.Tag}}' " + image
```
2. This label is added in slave.mk. See https://github.com/sonic-net/sonic-buildimage/blob/736c739bf492a73c60cfca000b853c040f53a104/slave.mk#L895 
```
	docker build --squash --no-cache \
...
		--label Tag=$(SONIC_IMAGE_VERSION) \
```
3. The real docker tag is added in build_debian.sh. See https://github.com/sonic-net/sonic-buildimage/blob/736c739bf492a73c60cfca000b853c040f53a104/files/build_templates/sonic_debian_extension.j2#L697
```
sudo LANG=C DOCKER_HOST="$DOCKER_HOST" chroot $FILESYSTEM_ROOT docker tag {{imagename}}:latest {{imagename}}:"${SONIC_IMAGE_VERSION}"
```

In theory, item#2 and item#3 shall get the same value of ${SONIC_IMAGE_VERSION}. But it doesn't. I got from build log:
1. For item#2, it prints:
```
docker-database.gz.log:Step 23/25 : LABEL Tag=syslog-rate-limit.0-dirty-20220804.115836
```
2. For item#3, it prints:
```
sonic-mellanox.bin.log:+ sudo LANG=C DOCKER_HOST= chroot ./fsroot-mellanox docker tag docker-database:latest docker-database:syslog-rate-limit.0-dirty-20220804.165951
```

As you can see, value of SONIC_IMAGE_VERSION is different in the same build. I suppose SONIC_IMAGE_VERSION  value shall always be the same in the same build.